java/core/src/java/org/apache/orc/Reader.java

```diff
@@ -237,6 +241,12 @@ public Options schema(TypeDescription schema) {
       return this;
     }
 
+    public Options setFilter(String[] filterColumnNames, Consumer<VectorizedRowBatch> filterCallback) {
```

Comment:
As discussed, this makes a lot of sense as it will enable us skipping multiple VectorBatches in a single call when there is no much -- as ORC-620 is making the filter API more generic I propose to tackle the issue there.